java.lang.NoClassDefFoundError: TextMateFile

I get the following error sporadically, when opening/switching projects:

```
java.lang.NoClassDefFoundError: org/jetbrains/plugins/textmate/psi/TextMateFile
	at com.redhat.devtools.lsp4ij.features.codeBlockProvider.LSPCodeBlockUtils.lambda$getBracePairsFwd$0(LSPCodeBlockUtils.java:112)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:173)
	at com.intellij.psi.impl.AbstractPsiCachedValue.doCompute(PsiCachedValueImpl.kt:21)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$3(CachedValueBase.java:299)
	at com.intellij.util.CachedValueBase.computeData(CachedValueBase.java:37)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$4(CachedValueBase.java:299)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:113)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:29)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:300)
	at com.intellij.psi.impl.AbstractPsiCachedValue.getValue(PsiCachedValueImpl.kt:15)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:83)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:170)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:136)
	at com.redhat.devtools.lsp4ij.features.codeBlockProvider.LSPCodeBlockUtils.getBracePairsFwd(LSPCodeBlockUtils.java:111)
	at com.redhat.devtools.lsp4ij.features.codeBlockProvider.LSPCodeBlockUtils.isCodeBlockStartChar(LSPCodeBlockUtils.java:62)
	at com.redhat.devtools.lsp4ij.features.codeBlockProvider.LSPCodeBlockProvider.getCodeBlockRange(LSPCodeBlockProvider.java:59)
	at com.redhat.devtools.lsp4ij.features.selectionRange.LSPCodeBlockExtendWordSelectionHandler.select(LSPCodeBlockExtendWordSelectionHandler.java:65)
	at com.intellij.codeInsight.editorActions.SelectWordUtil.askSelectioner(SelectWordUtil.java:229)
	at com.intellij.codeInsight.editorActions.SelectWordUtil.processElement(SelectWordUtil.java:209)
	at com.intellij.codeInsight.editorActions.SelectWordUtil.lambda$processInFile$1(SelectWordUtil.java:184)
	at com.intellij.openapi.project.DumbService.withAlternativeResolveEnabled(DumbService.kt:339)
	at com.intellij.codeInsight.editorActions.SelectWordUtil.processInFile(SelectWordUtil.java:181)
	at com.intellij.codeInsight.editorActions.SelectWordUtil.processRanges(SelectWordUtil.java:151)
	at com.intellij.codeInsight.editorActions.SelectWordHandler.doSelectWord(SelectWordHandler.java:172)
	at com.intellij.codeInsight.editorActions.SelectWordHandler.lambda$selectWord$0(SelectWordHandler.java:91)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:314)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:262)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:864)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:66)
	at com.intellij.codeInsight.editorActions.SelectWordHandler.lambda$selectWord$1(SelectWordHandler.java:90)
	at com.intellij.openapi.progress.impl.CoreProgressManager$1.run(CoreProgressManager.java:313)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:491)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:133)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$10(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:85)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:249)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:98)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:223)
	at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.use(trace.kt:45)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:222)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$14(CoreProgressManager.java:674)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:749)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:705)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:673)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:203)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:98)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:249)
	at com.intellij.openapi.progress.impl.ProgressRunner$ProgressRunnable.run(ProgressRunner.java:502)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:467)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:101)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:101)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:107)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:101)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$19(ProgressRunner.java:463)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:735)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:732)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:732)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.ClassNotFoundException: org.jetbrains.plugins.textmate.psi.TextMateFile PluginClassLoader(plugin=PluginDescriptor(name=LSP4IJ, id=com.redhat.devtools.lsp4ij, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/IntelliJIdea2024.3/LSP4IJ, version=0.10.0-20250207-202407, package=null, isBundled=false), packagePrefix=null, state=active, parents=PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.impl, descriptorPath=intellij.platform.vcs.impl.xml, path=/opt/intellij/idea-IU-242.20224.300/lib, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.log.impl, descriptorPath=intellij.platform.vcs.log.impl.xml, path=/opt/intellij/idea-IU-242.20224.300/lib, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.dvcs.impl, descriptorPath=intellij.platform.vcs.dvcs.impl.xml, path=/opt/intellij/idea-IU-242.20224.300/lib, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.collaborationTools, descriptorPath=intellij.platform.collaborationTools.xml, path=/opt/intellij/idea-IU-242.20224.300/lib, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=JSON, id=com.intellij.modules.json, descriptorPath=plugin.xml, path=/opt/intellij/idea-IU-242.20224.300/plugins/json, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=Telemetry by Red Hat, id=com.redhat.devtools.intellij.telemetry, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/IntelliJIdea2024.3/Telemetry by Red Hat, version=1.2.1.62, package=null, isBundled=false), )
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.kt:157)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 63 more
```

Plugin version: `0.10.0-20250207-202407`

> I get the following error sporadically, when opening/switching projects:
> 
> ```
> java.lang.NoClassDefFoundError: org/jetbrains/plugins/textmate/psi/TextMateFile
> 	at com.redhat.devtools.lsp4ij.features.codeBlockProvider.LSPCodeBlockUtils.lambda$getBracePairsFwd$0(LSPCodeBlockUtils.java:112)
> 	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:173)
> 	at com.intellij.psi.impl.AbstractPsiCachedValue.doCompute(PsiCachedValueImpl.kt:21)
> 	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$3(CachedValueBase.java:299)
> 	at com.intellij.util.CachedValueBase.computeData(CachedValueBase.java:37)
> 	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$4(CachedValueBase.java:299)
> 	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:113)
> 	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:29)
> 	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
> 	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:300)
> 	at com.intellij.psi.impl.AbstractPsiCachedValue.getValue(PsiCachedValueImpl.kt:15)
> 	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:83)
> 	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:170)
> 	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:136)
> 	at com.redhat.devtools.lsp4ij.features.codeBlockProvider.LSPCodeBlockUtils.getBracePairsFwd(LSPCodeBlockUtils.java:111)
> 	at com.redhat.devtools.lsp4ij.features.codeBlockProvider.LSPCodeBlockUtils.isCodeBlockStartChar(LSPCodeBlockUtils.java:62)
> 	at com.redhat.devtools.lsp4ij.features.codeBlockProvider.LSPCodeBlockProvider.getCodeBlockRange(LSPCodeBlockProvider.java:59)
> 	at com.redhat.devtools.lsp4ij.features.selectionRange.LSPCodeBlockExtendWordSelectionHandler.select(LSPCodeBlockExtendWordSelectionHandler.java:65)
> 	at com.intellij.codeInsight.editorActions.SelectWordUtil.askSelectioner(SelectWordUtil.java:229)
> 	at com.intellij.codeInsight.editorActions.SelectWordUtil.processElement(SelectWordUtil.java:209)
> 	at com.intellij.codeInsight.editorActions.SelectWordUtil.lambda$processInFile$1(SelectWordUtil.java:184)
> 	at com.intellij.openapi.project.DumbService.withAlternativeResolveEnabled(DumbService.kt:339)
> 	at com.intellij.codeInsight.editorActions.SelectWordUtil.processInFile(SelectWordUtil.java:181)
> 	at com.intellij.codeInsight.editorActions.SelectWordUtil.processRanges(SelectWordUtil.java:151)
> 	at com.intellij.codeInsight.editorActions.SelectWordHandler.doSelectWord(SelectWordHandler.java:172)
> 	at com.intellij.codeInsight.editorActions.SelectWordHandler.lambda$selectWord$0(SelectWordHandler.java:91)
> 	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:314)
> 	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:262)
> 	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:864)
> 	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:66)
> 	at com.intellij.codeInsight.editorActions.SelectWordHandler.lambda$selectWord$1(SelectWordHandler.java:90)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager$1.run(CoreProgressManager.java:313)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:491)
> 	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:133)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$10(CoreProgressManager.java:581)
> 	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:85)
> 	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:249)
> 	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:98)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:223)
> 	at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.use(trace.kt:45)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:222)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$14(CoreProgressManager.java:674)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:749)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:705)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:673)
> 	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:203)
> 	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:98)
> 	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:249)
> 	at com.intellij.openapi.progress.impl.ProgressRunner$ProgressRunnable.run(ProgressRunner.java:502)
> 	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:467)
> 	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:101)
> 	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:101)
> 	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:107)
> 	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:101)
> 	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$19(ProgressRunner.java:463)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
> 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:735)
> 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:732)
> 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
> 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:732)
> 	at java.base/java.lang.Thread.run(Thread.java:1583)
> Caused by: java.lang.ClassNotFoundException: org.jetbrains.plugins.textmate.psi.TextMateFile PluginClassLoader(plugin=PluginDescriptor(name=LSP4IJ, id=com.redhat.devtools.lsp4ij, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/IntelliJIdea2024.3/LSP4IJ, version=0.10.0-20250207-202407, package=null, isBundled=false), packagePrefix=null, state=active, parents=PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.impl, descriptorPath=intellij.platform.vcs.impl.xml, path=/opt/intellij/idea-IU-242.20224.300/lib, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.log.impl, descriptorPath=intellij.platform.vcs.log.impl.xml, path=/opt/intellij/idea-IU-242.20224.300/lib, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.dvcs.impl, descriptorPath=intellij.platform.vcs.dvcs.impl.xml, path=/opt/intellij/idea-IU-242.20224.300/lib, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.collaborationTools, descriptorPath=intellij.platform.collaborationTools.xml, path=/opt/intellij/idea-IU-242.20224.300/lib, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=JSON, id=com.intellij.modules.json, descriptorPath=plugin.xml, path=/opt/intellij/idea-IU-242.20224.300/plugins/json, version=243.23654.189, package=null, isBundled=true), PluginDescriptor(name=Telemetry by Red Hat, id=com.redhat.devtools.intellij.telemetry, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/IntelliJIdea2024.3/Telemetry by Red Hat, version=1.2.1.62, package=null, isBundled=false), )
> 	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.kt:157)
> 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
> 	... 63 more
> ```
> 
> Plugin version: `0.10.0-20250207-202407`

@SCWells72 is there any chance that you provide a fix please?

--- a/src/main/java/com/redhat/devtools/lsp4ij/LSPIJTextMateUtils.java
+++ b/src/main/java/com/redhat/devtools/lsp4ij/LSPIJTextMateUtils.java
@@ -11,6 +11,8 @@
 
 package com.redhat.devtools.lsp4ij;
 
+import com.intellij.lang.Language;
+import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.psi.PsiFile;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
@@ -24,8 +26,41 @@ import java.util.Map;
 @ApiStatus.Internal
 public final class LSPIJTextMateUtils {
 
+    private static final String TEXTMATE_LANGUAGE_ID = "textmate";
+
     private LSPIJTextMateUtils() {
         // Pure utility class
+    }
+
+
+    /**
+     * Returns the TextMate language and null otherwise.
+     *
+     * @returnthe TextMate language and null otherwise.
+     */
+    @Nullable
+    public static Language getTextMateLanguage() {
+        return Language.findLanguageByID(TEXTMATE_LANGUAGE_ID);
+    }
+
+    /**
+     * Returns true if the given file is a TextMate file and false otherwise.
+     *
+     * @param file the file.
+     * @return true if the given file is a TextMate file and false otherwise.
+     */
+    public static boolean isTextMateFile(@NotNull PsiFile file) {
+        return TEXTMATE_LANGUAGE_ID.equals(file.getLanguage().getID());
+    }
+
+    /**
+     * Returns true if the given file is a TextMate file and false otherwise.
+     *
+     * @param file the file.
+     * @return true if the given file is a TextMate file and false otherwise.
+     */
+    public static boolean isTextMateFile(@NotNull VirtualFile file) {
+        return TEXTMATE_LANGUAGE_ID.equals(file.getFileType().getName());
     }
 
     /**

--- a/src/main/java/com/redhat/devtools/lsp4ij/dap/client/variables/providers/TextMateVariableRangeRegistrar.java
+++ b/src/main/java/com/redhat/devtools/lsp4ij/dap/client/variables/providers/TextMateVariableRangeRegistrar.java
@@ -15,6 +15,7 @@ import com.intellij.openapi.project.Project;
 import com.intellij.openapi.util.TextRange;
 import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.psi.tree.IElementType;
+import com.redhat.devtools.lsp4ij.LSPIJTextMateUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.plugins.textmate.language.syntax.lexer.TextMateElementType;
 import org.jetbrains.plugins.textmate.language.syntax.lexer.TextMateScope;
@@ -33,7 +34,7 @@ public class TextMateVariableRangeRegistrar implements VariableRangeRegistrar {
 
     @Override
     public boolean isApplicable(@NotNull VirtualFile file, @NotNull Project project) {
-        return "textmate".equals(file.getFileType().getName());
+        return LSPIJTextMateUtils.isTextMateFile(file);
     }
 
     @Override

--- a/src/main/java/com/redhat/devtools/lsp4ij/features/codeBlockProvider/LSPCodeBlockUtils.java
+++ b/src/main/java/com/redhat/devtools/lsp4ij/features/codeBlockProvider/LSPCodeBlockUtils.java
@@ -22,7 +22,6 @@ import com.redhat.devtools.lsp4ij.LSPIJTextMateUtils;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
-import org.jetbrains.plugins.textmate.psi.TextMateFile;
 
 import java.util.HashMap;
 import java.util.Map;
@@ -109,7 +108,7 @@ public final class LSPCodeBlockUtils {
     @NotNull
     private static Map<Character, Character> getBracePairsFwd(@NotNull PsiFile file) {
         return CachedValuesManager.getCachedValue(file, () -> {
-            Map<Character, Character> bracePairs = file instanceof TextMateFile ? LSPIJTextMateUtils.getBracePairs(file) :
+            Map<Character, Character> bracePairs = LSPIJTextMateUtils.isTextMateFile(file) ? LSPIJTextMateUtils.getBracePairs(file) :
                     file.getFileType() instanceof AbstractFileType ? getAbstractFileTypeBracePairs(file) :
                             null;
             if (bracePairs == null) {

--- a/src/main/java/com/redhat/devtools/lsp4ij/internal/SimpleLanguageUtils.java
+++ b/src/main/java/com/redhat/devtools/lsp4ij/internal/SimpleLanguageUtils.java
@@ -12,6 +12,7 @@ package com.redhat.devtools.lsp4ij.internal;
 
 import com.intellij.lang.Language;
 import com.intellij.openapi.fileTypes.PlainTextLanguage;
+import com.redhat.devtools.lsp4ij.LSPIJTextMateUtils;
 
 import java.util.Collections;
 import java.util.HashSet;
@@ -29,7 +30,7 @@ public class SimpleLanguageUtils {
         // plain/text
         languages.add(PlainTextLanguage.INSTANCE);
         // textmate
-        Language textMateLanguage = Language.findLanguageByID("textmate");
+        Language textMateLanguage = LSPIJTextMateUtils.getTextMateLanguage();
         if (textMateLanguage != null) {
             languages.add(textMateLanguage);
         }

